tests/nested/manual/core20-install-device-file-install-via-hook-hack/task.yaml

```diff
@@ -103,7 +103,6 @@ prepare: |
 
     # start fake device svc
     systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
-    tests.cleanup defer systemctl stop fakedevicesvc
```

Comment:
hmm I'm confused by this change, why does doing it this way not work?

Reply:
@anonymouse64, sorry, I didn't explain that in the pr, the problem here is that "tests.cleanup restore" is executed on
"$TESTSLIB"/prepare-restore.sh --restore-suite-each, but nested tests don't run this because to restore a nested test we just need to delete the VM and clean the test directories. 

So for nested tests we don't have to use "tests.cleanup defer" at lease until we add that to the tests restore phase.

Reply:
otherwise, I could add "tests.cleanup restore" for all the nested suites

Reply:
nice, thank you for making that change so that `tests.cleanup defer` works again